javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -821,6 +821,9 @@ private void addModelConstructor(ClientModel model, JavaVisibility constructorVi
                 if (modelDefinesProperty(model, polymorphicProperty)) {
                     constructor.line("this." + polymorphicProperty.getName() + " = " + discriminatorValue + ";");
                 }
+                if (ClientModelUtil.isJsonMergePatchModel(model)) {
+                    constructor.line(String.format("this.updatedProperties.add(\"%s\");", polymorphicProperty.getName()));
```

Comment:
nit: avoid using `String.format`. It's performance is very bad. We can just use string concatenation.

```suggestion
                    constructor.line("this.updatedProperties.add(\"" + polymorphicProperty.getName() + "\");");
```

Reply:
Fixed for the two prs on json merge patch .